lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -2264,7 +2264,8 @@ CompilerType TypeSystemSwiftTypeRef::GetChildCompilerTypeAtIndex(
   // Because the API deals out an index into a list of children we
   // can't mix&match between the two typesystems if there is such a
   // divergence. We'll need to replace all calls at once.
-  if (m_swift_ast_context->GetNumFields(ReconstructType(type)) <
+  if (m_swift_ast_context->GetNumChildren(ReconstructType(type),
```

Comment:
is `ReconstructType` needed here? Why can't this call be passed `type` directly.

Reply:
type is an opaque pointer in TypeSystemSwiftTypeRef (a mangled name). If we want something to pass to SwiftASTContext we need to convert it to an AST type first.